opds2_import.py

```diff
@@ -0,0 +1,830 @@
+
+        :return: Publication's metadata
+        :rtype: Metadata
+        """
+        self._logger.debug(
+            u"Started extracting metadata from publication {0}".format(
+                encode(publication)
+            )
+        )
+
+        title = publication.metadata.title
+
+        if title == OPDSFeed.NO_TITLE:
+            title = None
+
+        subtitle = publication.metadata.subtitle
+
+        languages = first_or_default(publication.metadata.languages)
+        derived_medium = self._extract_medium_from_links(publication.links)
+        medium = self._extract_medium(publication, derived_medium)
+
+        publisher = first_or_default(publication.metadata.publishers)
+        if publisher:
+            publisher = publisher.name
+
+        published = publication.metadata.published
+        subjects = self._extract_subjects(publication.metadata.subjects)
+        contributors = (
+            self._extract_contributors(publication.metadata.authors)
+            + self._extract_contributors(publication.metadata.translators)
+            + self._extract_contributors(publication.metadata.editors)
+            + self._extract_contributors(publication.metadata.artists)
+            + self._extract_contributors(publication.metadata.illustrators)
+            + self._extract_contributors(publication.metadata.letterers)
+            + self._extract_contributors(publication.metadata.pencilers)
+            + self._extract_contributors(publication.metadata.colorists)
+            + self._extract_contributors(publication.metadata.inkers)
+            + self._extract_contributors(publication.metadata.narrators)
+            + self._extract_contributors(publication.metadata.contributors)
+            + self._extract_contributors(publication.metadata.imprints)
+        )
+
+        feed_self_url = first_or_default(
+            feed.links.get_by_rel(OPDS2LinkRelationsRegistry.SELF.key)
+        ).href
+        links = self._extract_links(publication, feed_self_url)
+
+        last_opds_update = publication.metadata.modified
+
+        identifier, _ = Identifier.parse_urn(self._db, publication.metadata.identifier)
+        identifier_data = IdentifierData(
+            type=identifier.type, identifier=identifier.identifier
+        )
+
+        # FIXME: There are no measurements in OPDS 2.0
+        measurements = []
+
+        # FIXME: There is no series information in OPDS 2.0
+        series = None
+        series_position = None
+
+        # FIXME: It seems that OPDS 2.0 spec doesn't contain information about rights so we use the default one
+        rights_uri = RightsStatus.rights_uri_from_string("")
+
+        circulation_data = CirculationData(
+            default_rights_uri=rights_uri,
+            data_source=data_source_name,
+            primary_identifier=identifier_data,
+            links=links,
+            licenses_owned=LicensePool.UNLIMITED_ACCESS,
+            licenses_available=LicensePool.UNLIMITED_ACCESS,
```

Comment:
Up to this point open-access books have had licenses_owned=0 and licenses_available=0. I agree that now that we have UNLIMITED_ACCESS, it makes more sense to use that instead. The only place I know this will cause problems is in the dashboard, where we already know we have a problem with the -1 value for unlimited-access books (I filed https://jira.nypl.org/browse/SIMPLY-3238 for this; it's not urgent).

In the future I'd like to see this value standardized for open-access works obtained through other means; this would mean changes to other modules + a migration script.